pkg/tfvars/internal/cache/cache.go

```diff
@@ -193,9 +198,90 @@ func DownloadFile(baseURL string, dataType string) (string, error) {
 }
 
 // DownloadImageFile is a helper function that obtains an image file from a given URL,
-// puts it in the cache and returns the local file path.
+// puts it in the cache and returns the local file path.  If the file is compressed
+// by a known compressor, the file is uncompressed prior to being returned.
 func DownloadImageFile(baseURL string) (string, error) {
 	logrus.Infof("Obtaining RHCOS image file from '%v'", baseURL)
 
 	return DownloadFile(baseURL, imageDataType)
 }
+
+func detectContentType(file string) (string, error) {
+	logrus.Debugf("Detecting content type for %s", file)
+	fd, err := os.Open(file)
+	if err != nil {
+		return "", err
+	}
+	defer fd.Close()
+
+	// See http://golang.org/pkg/net/http/#DetectContentType for why we use 512
+	buf := make([]byte, 512)
+	_, err = fd.Read(buf)
+	if err != nil {
+		return "", err
+	}
+	filetype := http.DetectContentType(buf)
+	logrus.Debugf("detectContentType %s is %s", file, filetype)
+	return filetype, nil
+}
+
+func decompressFileGzip(src, dest string) error {
+	logrus.Debugf("decompressFileGzip %s %s", src, dest)
+	gzipfile, err := os.Open(src)
+	if err != nil {
+		return err
+	}
+
+	reader, err := gzip.NewReader(gzipfile)
+	if err != nil {
+		return err
+	}
+	defer reader.Close()
+
+	writer, err := os.Create(dest)
+	if err != nil {
+		return err
+	}
+	defer writer.Close()
+
+	if _, err = io.Copy(writer, reader); err != nil {
+		os.Remove(dest)
+		return err
+	}
+
+	return nil
+}
+
+// decompressFile attempts to decompress a file in the cache if
+// appropriate, based on the detected content type.  If the
+// file type is not known, it simply leaves the file untouched.
+func decompressFile(src, dest string) error {
+	filetype, err := detectContentType(src)
+	if err != nil {
+		return err
+	}
+	if decompressor, exists := compressorMap[filetype]; exists {
+		if src == dest {
+			tmpSrc := src + ".tmp"
+			err = os.Rename(src, tmpSrc)
+			if err != nil {
+				return err
+			}
+			src = tmpSrc
+		}
+		uncompressTmp := src + "_uncompressed"
+		err = decompressor(src, uncompressTmp)
+		if err != nil {
+			return err
+		}
+		err = os.Remove(src)
+		if err != nil {
+			return err
+		}
+		err = os.Rename(uncompressTmp, src)
```

Comment:
are both of these required?

Reply:
It should.  I was debating (briefly) what to do, but I'm happy enough to change it.

Reply:
Obsolete.